src/views/MobileAccount.vue

```diff
@@ -229,53 +228,50 @@ export default class MobileAccount extends Vue {
     await this.checkPendingWithdrawalReceipt()
     await this.checkUnclaimedLoomTokens()
     this.currentAllowance = await this.checkAllowance()
-    this.queryRewards()
     this.updateTimeUntilElectionCycle()
     this.startTimer()
-    this.delegations = await this.getDelegations()
   }
 
   refresh() {
     this.showRefreshSpinner = true
-    this.$root.$emit("refreshBalances")
-    setTimeout(() => {
-      this.showRefreshSpinner = false
-    }, 2000)
-  }
-
-  async getDelegations() {
-    const { amount, weightedAmount, delegationsArray } = await this.dposUser.listDelegatorDelegations()
-    const candidates = await this.dposUser.listCandidatesAsync()
-
-    return delegationsArray.filter(d => !(d.amount.isZero() && d.updateAmount.isZero()))
-           .map(delegation => {
-            let candidate = candidates.find(c => c.address.local.toString() === delegation.validator.local.toString())
-            return { 
-                    "Name": candidate.name,
-                    "Amount": `${formatToCrypto(delegation.amount)}`,
-                    "Update Amount": `${formatToCrypto(delegation.updateAmount)}`,
-                    "Height": `${delegation.height}`,
-                    "Locktime": `${new Date(delegation.lockTime * 1000)}`,
-                    "State": `${this.states[delegation.state]}`,
-                    _cellVariants: { Status: 'active'}
-            }
+    Promise.all([
+        this.getMetamaskLoomBalance(),
+        this.getDappchainLoomBalance()
+    ])
+    .finally(() => this.showRefreshSpinner = false)
+  }
+
+  get formatedDelegations() {
+    const candidates = this.validators
+    console.log(this.delegations)
+    return this.delegations
+    .filter(d => d.index > 0)
+    .map((delegation) => {
+      let candidate = candidates.find(c => c.address === delegation.validator.local.toString())
+      return { 
+              "Name": candidate.name,
+              "Amount": `${formatToCrypto(delegation.amount)}`,
+              "Update Amount": `${formatToCrypto(delegation.updateAmount)}`,
+              "Locktime": `${new Date(delegation.lockTime * 1000)}`,
+              "State": `${this.states[delegation.state]}`,
+              _cellVariants: { Status: 'active'}
+      }
     })
-
   }
 
   toggleAccordion(idx) {
     this.$root.$emit("bv::toggle::collapse", "accordion" + idx)
   }
 
   async completeDeposit() {
+    const dposUser = await this.dposUser
     this.setGatewayBusy(true)
     this.setShowLoadingSpinner(true)
     const tokens = new BN( "" + parseInt(this.currentAllowance,10)) 
     const weiAmount = new BN(this.web3.utils.toWei(tokens, 'ether'), 10)
```

Comment:
We should eventually start converting from `web3.utils.toWei` to `ethers.utils.bigNumberify`, but that would require a few changes in loom-js so that it accepts ethers.

If you have a number in Wei that you want to display in Ether, (ie divide by 1e18), use `etherString = ethers.utils.formatEther(balance);`

(this comment is informational)

Reply:
Yes. Did just that any a bunch of places that were on my radar. We will do it everywhere. Ultimately the only place we should see web3 is in wallet connection related code.